packages/shell-api/src/stream-processor.ts

```diff
@@ -12,18 +12,68 @@ import {
 import type { Streams } from './streams';
 import type { MQLPipeline } from './mql-types';
 
+export type StreamProcessorData = Document & { name: string };
+
 @shellApiClassDefault
 export class StreamProcessor extends ShellApiWithMongoClass {
-  constructor(public _streams: Streams, public name: string) {
+  public name: string;
+  public id?: string;
+  public pipeline?: MQLPipeline;
+  public state?: string;
+  public tier?: string;
+  public errorMsg?: string;
+  public lastModified?: Date;
+  public lastStateChange?: Date;
+
+  constructor(public _streams: Streams, data: StreamProcessorData) {
```

Comment:
Let's also make `_streams` a normal private property instead of using the constructor shorthand (moving closer to [`--erasableSyntaxOnly`](https://www.totaltypescript.com/erasable-syntax-only))